### ISSUE TITLE
windows pod error while adding to cni network: error while ProvisionEndpoint: Cannot create a file when that file already exists.

### ISSUE THREAD

**Issue by yuwei1chen** [kubernetes/kubernetes#116697]
### What happened?

Everything works fine for windows isolation mode pods in kubernetes.But for windows hyperv isolation mode pods, they have no pod ip and keeps restarting. 
below is what I got from "kubelet get pod -o wide", we can see that in IP column is "<none>". And it keeps restarting and have restarted 51 times:
`hyperv-0   0/1     CrashLoopBackOff   51         110m   <none>          jf53120cp-hv `

In kubelet log, we see:
`E0315 00:04:48.592295   11324 cni.go:366] Error adding hyperv-0/ded8068e29d711f9e996fdc753a99a1eb421509decb0035709f6e58291be8a23 to network flannel/flannel.4096: error while ProvisionEndpoint(ded8068e29d711f9e996fdc753a99a1eb421509decb0035709f6e58291be8a23_flannel.4096,D3CBB652-3DCC-486F-BBB5-690D54C96F18,ded8068e29d711f9e996fdc753a99a1eb421509decb0035709f6e58291be8a23): Cannot create a file when that file already exists.
E0315 00:04:48.592295   11324 cni_windows.go:59] error while adding to cni network: error while ProvisionEndpoint(ded8068e29d711f9e996fdc753a99a1eb421509decb0035709f6e58291be8a23_flannel.4096,D3CBB652-3DCC-486F-BBB5-690D54C96F18,ded8068e29d711f9e996fdc753a99a1eb421509decb0035709f6e58291be8a23): Cannot create a file when that file already exists.`

The only different in 2 isolation mode pods is the annotation "experimental.windows.kubernetes.io/isolation-type: hyperv", which is to tell kubelet which windows container isolation mode to use.

### What did you expect to happen?

We want to run docker windows hyperv isolation mode in kubernetes.
Both Windows hyperv isolation mode and process isolation mode container pod in kubernetes is started correctly with a pod ip and right network configuration.

### How can we reproduce it (as minimally and precisely as possible)?

Set up a v1.20 k8s cluster and join Windows server 2019 worker node. enable hyperv isolation mode feature gate in kubelet.
And start any hyperv isolation mode pod with annotation "experimental.windows.kubernetes.io/isolation-type: hyperv".

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
PS D:\testagents\dockerfile> wmic os get Caption
Caption
Microsoft Windows Server 2019 Datacenter

PS D:\testagents\dockerfile> wmic os get Version
Version
10.0.17763

PS D:\testagents\dockerfile> wmic os get BuildNumber
BuildNumber
17763

PS D:\testagents\dockerfile> wmic os get OSArchitecture
OSArchitecture
64-bit
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
Docker v20.10.9
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>
Flannel flannel:v0.13.0-nanoserver
</details>

**Comment by feiskyer** [comment]
The error is coming from flannel. Have you already filed an issue to Flannel?

**Comment by yuwei1chen** [comment]
> The error is coming from flannel. Have you already filed an issue to Flannel?

Yes. We have filed https://github.com/flannel-io/flannel/issues/1742.
BTW, we know that this feature gate was developed by you, could you share when you tested this feature then, which CNI were you using?

**Comment by yuwei1chen** [comment]
After keep creating hyperv isolation mode pods for a weekend, new pods cannot be created with message:
`Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "80da9792a7eeaf1b15f4ffc60cfc8fc24942b540b2be8df4898a445b13377409" network for pod "kube-proxy-windows-k5m76": networkPlugin cni failed to set up pod "kube-proxy-windows-k5m76_kube-system" network: error while ProvisionEndpoint(80da9792a7eeaf1b15f4ffc60cfc8fc24942b540b2be8df4898a445b13377409_flannel.4096,837E2676-2E40-475C-B8A0-6961B7BE823B,80da9792a7eeaf1b15f4ffc60cfc8fc24942b540b2be8df4898a445b13377409): failed to make a new HNSEndpoint: error while ipam.ExecAdd: failed to allocate for range 0: no IP addresses available in range set: 10.244.24.1-10.244.24.254`

On this host, I found that in "C:\var\lib\cni\netwokrs\flannel.4096" all ip addresses are taken from 10.244.24.2 to 10.244.24.254 but there are no pod running on this server except flannel. It seems that ip addresses were not released correctly. **Since everything works fine for process isolation mode pod, my guess is that there's something wrong with network destroying part for hyperv isolation mode pods.** What should be the cause, cni, flannel, kubelet or other components? @feiskyer

**Comment by yuwei1chen** [comment]
Is there any recommended doc for deploying cni on windows nodes?
